src/main/java/org/traccar/handler/FilterHandler.java

```diff
@@ -167,21 +168,43 @@ private boolean filter(Position position) {
         if (filterApproximate(position)) {
             filterType.append("Approximate ");
         }
-        if (filterStatic(position) && !skipLimit(position, last) && !skipAttributes(position)) {
+        if (filterStatic(position)) {
             filterType.append("Static ");
         }
-        if (filterDistance(position, last) && !skipLimit(position, last) && !skipAttributes(position)) {
-            filterType.append("Distance ");
-        }
-        if (filterMaxSpeed(position, last)) {
-            filterType.append("MaxSpeed ");
-        }
-        if (filterMinPeriod(position, last)) {
-            filterType.append("MinPeriod ");
+
+        // relative filtering
+        if (filterDuplicate || filterDistance != 0 || filterMaxSpeed != 0 || filterMinPeriod != 0) {
+            Position previous = null;
+            if (filterRelative) {
+                try {
+                    Date newFixTime = position.getFixTime();
+                    previous = Context.getDataManager().getPrevPosition(position.getDeviceId(), newFixTime);
+                } catch (SQLException e) {
+                    LOGGER.warn("Error filtering position", e);
```

Comment:
This should probably be some more relevant message, like "Error retrieving previous position" or something along those lines.

Reply:
We can also fallback to `getLastPosition`.